tests/integration/extenders/ModelTest.php

```diff
@@ -167,6 +168,66 @@ public function custom_relationship_is_inherited_to_child_classes()
         $this->assertEquals(1, $post->ancestor->id);
     }
 
+    /**
+     * @test
+     */
+    public function custom_relationship_prioritizes_child_classes_within_2_parent_classes()
+    {
+        $this->extend(
+            (new Extend\Model(AbstractModel::class))
+                ->belongsTo('ancestor', User::class, 'user_id'),
```

Comment:
What is the idea behind this scenario? I see no use-case for extending this abstract base class. And as long as no use-case pops up, I'd rather not define how we want to (and then have to) behave.

The `Post` / `CommentPost` scenario you added below makes sense. :+1: